src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -234,6 +246,18 @@ def transform_checkin_uuid(
     return check_in_guid, use_latest_checkin
 
 
+@dataclasses.dataclass(frozen=True)
+class ProcessingError:
+    type: ProcessingErrorType
+    data: dict[str, Any] = dataclasses.field(default_factory=dict)
+
+
+class CheckinValidationError(Exception):
+    def __init__(self, processing_errors: list[ProcessingError], monitor: Monitor | None = None):
+        self.processing_errors = processing_errors
+        self.monitor = monitor
```

Comment:
Why include the monitor sometimes again? Should we capture in a comment?

Reply:
We aren't always able to access it at the point that we failed, so sometimes we just have the slug